python/cudf/cudf/core/frame.py

```diff
@@ -304,6 +304,70 @@ def _unaryop(self, op):
             result._data[name] = col.unary_operator(op)
         return result
 
+    def interleave_columns(self):
+        """
+       Interleave Series columns of a table into a single column.
+
+        Converts the column major table `cols` into a row major column.
+        Parameters
+        ----------
+        cols : input Table containing columns to interleave.
+
+        Example
+        -------
+        >>> df = DataFrame([['A1', 'A2', 'A3'], ['B1', 'B2', 'B3']])
+        >>> df
+        0    [A1, A2, A3]
+        1    [B1, B2, B3]
+        >>> df.interleave_columns()
+        0    A1
+        1    B1
+        2    A2
+        3    B2
+        4    A3
+        5    B3
+
+        Returns
+        -------
+        The interleaved columns as a single column
+        """
+
+        return self._constructor_sliced(
+            libcudfxx.reshape.interleave_columns(self)
+        )
+
+    def tile(self, count):
+        """
+        Repeats the rows from `self` DataFrame `count` times to form a
+        new DataFrame.
+
+        Different from DataFrame.tile() which repeats the DataFrame
+        sequentially with the filling function. This repeats the
+        columns by "count" number of times.
+
+        Parameters
+        ----------
+        self : input Table containing columns to interleave.
+        count : Number of times to tile "rows". Must be non-negative.
+
+        Example
+        -------
+        >>> df  = Dataframe([[8, 4, 7], [5, 2, 3]])
+        >>> count = 2
+        >>> df.tile(df, count)
+           0  1  2  3  4  5
+        0  8  4  7  8  4  7
+        1  5  2  3  5  2  3
+
+        Returns
+        -------
+        The table containing the tiled "rows".
+        """
+        result = self.__class__._from_table(
+            libcudfxx.reshape.tile(self, count)
+        )
+        return result
```

Comment:
I think we need to do `result._copy_categories(self)` to handle setting the categories properly on the output

Reply:
ooooh. That is a missing part of the puzzle. Thank you.